tests/helpers/test_network.py

```diff
@@ -185,82 +177,73 @@ async def test_get_url_external(hass: HomeAssistant):
+    assert _get_external_url(hass) == "http://example.com:8123"
+    assert _get_external_url(hass, allow_cloud=False) == "http://example.com:8123"
+    assert _get_external_url(hass, allow_ip=False) == "http://example.com:8123"
+    assert _get_external_url(hass, prefer_cloud=True) == "http://example.com:8123"
 
     with pytest.raises(NoURLAvailableError):
-        _async_get_external_url(hass, require_standard_port=True)
+        _get_external_url(hass, require_standard_port=True)
 
     with pytest.raises(NoURLAvailableError):
-        _async_get_external_url(hass, require_ssl=True)
+        _get_external_url(hass, require_ssl=True)
 
     # Test with external URL: http://example.com:80/
     await async_process_ha_core_config(
         hass, {"external_url": "http://example.com:80/"},
     )
 
     assert hass.config.external_url == "http://example.com:80/"
-    assert _async_get_external_url(hass) == "http://example.com"
-    assert _async_get_external_url(hass, allow_cloud=False) == "http://example.com"
-    assert _async_get_external_url(hass, allow_ip=False) == "http://example.com"
-    assert _async_get_external_url(hass, prefer_cloud=True) == "http://example.com"
-    assert (
-        _async_get_external_url(hass, require_standard_port=True)
-        == "http://example.com"
-    )
+    assert _get_external_url(hass) == "http://example.com"
+    assert _get_external_url(hass, allow_cloud=False) == "http://example.com"
+    assert _get_external_url(hass, allow_ip=False) == "http://example.com"
+    assert _get_external_url(hass, prefer_cloud=True) == "http://example.com"
+    assert _get_external_url(hass, require_standard_port=True) == "http://example.com"
 
     with pytest.raises(NoURLAvailableError):
-        _async_get_external_url(hass, require_ssl=True)
+        _get_external_url(hass, require_ssl=True)
 
     # Test with external url: https://example.com:443/
     await async_process_ha_core_config(
         hass, {"external_url": "https://example.com:443/"},
     )
     assert hass.config.external_url == "https://example.com:443/"
-    assert _async_get_external_url(hass) == "https://example.com"
-    assert _async_get_external_url(hass, allow_cloud=False) == "https://example.com"
-    assert _async_get_external_url(hass, allow_ip=False) == "https://example.com"
-    assert _async_get_external_url(hass, prefer_cloud=True) == "https://example.com"
-    assert _async_get_external_url(hass, require_ssl=False) == "https://example.com"
-    assert (
-        _async_get_external_url(hass, require_standard_port=True)
-        == "https://example.com"
-    )
+    assert _get_external_url(hass) == "https://example.com"
+    assert _get_external_url(hass, allow_cloud=False) == "https://example.com"
+    assert _get_external_url(hass, allow_ip=False) == "https://example.com"
+    assert _get_external_url(hass, prefer_cloud=True) == "https://example.com"
+    assert _get_external_url(hass, require_ssl=False) == "https://example.com"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)